src/journal.rs

```diff
@@ -309,7 +313,11 @@ impl Journal {
         let c: *mut c_char = ptr::null_mut();
         if unsafe { ffi::sd_journal_get_cursor(self.j, &c) != 0 } {
             // Cursor may need to be re-aligned on a real entry first.
-            sd_try!(ffi::sd_journal_next(self.j));
+            if tail {
+                sd_try!(ffi::sd_journal_previous(self.j));
```

Comment:
@jlebon did you run a `cargo test` after this change? On my local systemd (v239) this broke the tests:
```
test test_simple_match ... FAILED

---- test_simple_match stdout ----
thread 'test_simple_match' panicked at 'assertion failed: r.is_some()', tests/journal.rs:100:5
```

(and I think that travis does not have a recent enough journald running to check this)

Reply:
That's indeed the case. For some reason this just made the test quite reliably losing the race.